### PR TITLE
Deprecate negative predicates

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: https://jira.rax.io/browse/FRMW-915

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM
- [x] QE LGTM

### CODE REVIEW

File: utils/rx-page-objects/index.js [hunk 1]
```diff
@@ -3,27 +3,25 @@
...
+var pageObjects = {
+    exercise: {}
 };
 
 require('fs').readdirSync(srcPath).forEach(file => {
-    var destination = pageObjects;
-    if (/.*exercise\.js$/.test(file)) {
-        destination = exercises;
+    // ignore unless JS
+    if (!/.js$/.test(file)) {
```
Comment (CITguy) [inline utils/rx-page-objects/index.js]: Note to Self: escape period `\.`

File: utils/rx-page-objects/src/rxCheckbox.page.js [hunk 2]
```diff
@@ -50,21 +50,36 @@ var rxCheckbox = {
...
+        }
+    },
+
+    /**
+     * @instance
+     * @function
+     * @description Whether or not the checkbox is enabled.
+     * @returns {Promise<Boolean>}
+     */
+    isEnabled: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxCheckbox.page.js]: (Release Notes) ADDED `rxCheckbox.isEnabled()`

File: utils/rx-page-objects/src/rxDatePicker.page.js [hunk 3]
```diff
@@ -120,17 +120,35 @@ var rxDatePicker = {
...
+        }
+    },
+
+    /**
+     * @function
+     * @instance
+     * @description Whether or not the entire calendar component is enabled.
+     * @return {Promise<Boolean>}
+     */
+    isEnabled: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxDatePicker.page.js]: (Release Notes) ADDED `rxDatePicker.isEnabled()`

File: utils/rx-page-objects/src/rxRadio.page.js [hunk 4]
```diff
@@ -76,21 +76,36 @@ var rxRadio = {
...
+        }
+    },
+
+    /**
+     * @function
+     * @instance
+     * @description Whether or not the radio element is enabled.
+     * @return {Promise<Boolean>}
+     */
+    isEnabled: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxRadio.page.js]: (Release Notes) Added `rxRadio.isEnabled()`

File: utils/rx-page-objects/src/rxSearchBox.page.js [hunk 5]
```diff
@@ -70,26 +70,43 @@ var rxSearchBox = {
...
+        }
+    },
+
+    /**
+     * @function
+     * @instance
+     * @description Whether or not the search box is enabled.
+     * @returns {Promise<Boolean>}
+     */
+    isEnabled: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxSearchBox.page.js]: ⬜ (Release Notes) ADDED `rxSearchBox.isEnabled()`

File: utils/rx-page-objects/src/rxSelect.page.js [hunk 6]
```diff
@@ -98,21 +98,36 @@ var rxSelect = {
...
+        }
+    },
+
+    /**
+     * @instance
+     * @function
+     * @description Whether or not the select element is enabled.
+     * @returns {Promise<Boolean>}
+     */
+    isEnabled: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxSelect.page.js]: ⬜ (Release Notes) ADDED: `rxSelect.isEnabled()`

File: utils/rx-page-objects/src/rxActionMenu.page.js [hunk 7]
```diff
@@ -104,8 +104,11 @@ var rxActionMenu = {
...
      * @function
      * @instance
+     * @deprecated
      * @description Whether or not the action cog is hiding its underlying menu.
-     * @returns {Boolean}
+     *
+     * **DEPRECATED** Check inverse of `isExpanded()` instead
+     * @returns {Promise<Boolean>}
      */
     isCollapsed: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxActionMenu.page.js]: ⬜ (Release Notes) DEPRECATED `rxActionMenu.isCollapsed()`. Use `rxActionMenu.isExpanded()`

File: utils/rx-page-objects/src/rxApp.page.js [hunk 8]
```diff
@@ -64,38 +64,41 @@ var rxApp = {
...
      * @function
-     * @returns {Boolean}
+     * @deprecated
      * @description Whether or not the navigation section is collapsed. If the navigation section is
      * not collapsible, it will throw an {@link rxApp#NotCollapsibleException}.
+     *
+     * **DEPRECATED** Check inverse of `isExpanded()` instead.
+     * @returns {Promise<Boolean>}
      */
     isCollapsed: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxApp.page.js]: ⬜ (Release Notes) DEPRECATED `rxApp.isCollapsed()`. Use `rxApp.isExpanded()` instead.

File: utils/rx-page-objects/src/rxCheckbox.page.js [hunk 9]
```diff
@@ -50,21 +50,36 @@ var rxCheckbox = {
...
      * @instance
      * @function
+     * @deprecated
      * @description Whether or not the checkbox is disabled.
-     * @returns {Boolean}
+     *
+     * **DEPRECATED** Check for inverse of `isEnabled()` instead.
+     * @returns {Promise<Boolean>}
      */
     isDisabled: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxCheckbox.page.js]: ⬜ (Release Notes) DEPRECATED `rxCheckbox.isDisabled()` Use `rxCheckbox.isEnabled()` instead.

File: utils/rx-page-objects/src/rxCollapse.page.js [hunk 10]
```diff
@@ -33,8 +33,11 @@ var rxCollapse = {
...
      * @instance
      * @function
+     * @deprecated
      * @description Whether or not the component is currently collapsed.
-     * @returns {Boolean}
+     *
+     * **DEPRECATED** Check inverse of `isExpanded()` instead.
+     * @returns {Promise<Boolean>}
      */
     isCollapsed: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxCollapse.page.js]: ⬜ (Release Notes) DEPRECATED `rxCollapse.isCollapsed()`. Use `rxCollapse.isExpanded()` instead.

File: utils/rx-page-objects/src/rxDatePicker.page.js [hunk 11]
```diff
@@ -120,17 +120,35 @@ var rxDatePicker = {
...
      * @function
      * @instance
+     * @deprecated
      * @description Whether or not the entire calendar component is disabled.
-     * @return {Boolean}
+     *
+     * **DEPRECATED** Check inverse of `isEnabled()` instead.
+     * @return {Promise<Boolean>}
      */
     isDisabled: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxDatePicker.page.js]: ⬜ (Release Notes) DEPRECATED `rxDatePicker.isDisabled()`. Use `rxDatePicker.isEnabled()` instead.

File: utils/rx-page-objects/src/rxDatePicker.page.js [hunk 12]
```diff
@@ -176,8 +194,11 @@ var rxDatePicker = {
...
      * @instance
      * @function
+     * @deprecated
      * @description Whether or not the calendar is closed.
-     * @return {Boolean}
+     *
+     * **DEPRECATED** Check inverse of `isOpen()` instead.
+     * @return {Promise<Boolean>}
      */
     isClosed: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxDatePicker.page.js]: ⬜ (Release Notes) DEPRECATED `rxDatePicker.isClosed()` Use `rxDatePicker.isOpen()` instead.

File: utils/rx-page-objects/src/rxRadio.page.js [hunk 13]
```diff
@@ -76,21 +76,36 @@ var rxRadio = {
...
      * @function
      * @instance
+     * @deprecated
      * @description Whether or not the radio element is disabled.
-     * @return {Boolean}
+     *
+     * **DEPRECATED** Check for inverse of `isEnabled()` instead.
+     * @return {Promise<Boolean>}
      */
     isDisabled: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxRadio.page.js]: ⬜ (Release Notes) DEPRECATED `rxRadio.isDisabled()` Use `rxRadio.isEnabled()` instead.

File: utils/rx-page-objects/src/rxSearchBox.page.js [hunk 14]
```diff
@@ -70,26 +70,43 @@ var rxSearchBox = {
...
      * @instance
+     * @deprecated
      * @description Whether or not the search box is searchable.
-     * To make this check, {@link rxSearchBox#isDisabled} is called.
-     * @returns {Boolean}
+     *
+     * **DEPRECATED** Use `isEnabled()` instead.
+     * @returns {Promise<Boolean>}
      */
     isSearchable: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxSearchBox.page.js]: ⬜ (Release Notes) DEPRECATED `rxSearchBox.isSearchable()` Use `rxSearchBox.isEnabled()` instead.

File: utils/rx-page-objects/src/rxSearchBox.page.js [hunk 15]
```diff
@@ -70,26 +70,43 @@ var rxSearchBox = {
...
      * @function
      * @instance
+     * @deprecated
      * @description Whether or not the search box is disabled.
-     * @returns {Boolean}
+     *
+     * **DEPRECATED** Check inverse of `isEnabled()` instead.
+     * @returns {Promise<Boolean>}
      */
     isDisabled: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxSearchBox.page.js]: ⬜ (Release Notes) DEPRECATED `rxSearchBox.isDisabled()` Use `rxSearchBox.isEnabled()` instead.

File: utils/rx-page-objects/src/rxSelect.page.js [hunk 16]
```diff
@@ -98,21 +98,36 @@ var rxSelect = {
...
      * @instance
      * @function
+     * @deprecated
      * @description Whether or not the select element is disabled.
-     * @returns {Boolean}
+     *
+     * **DEPRECATED** check for inverse of `isEnabled()` instead.
+     * @returns {Promise<Boolean>}
      */
     isDisabled: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxSelect.page.js]: ⬜ (Release Notes) DEPRECATED `rxSelect.isDisabled()` Use `rxSelect.isEnabled()` instead.

File: utils/rx-page-objects/src/rxTimePicker.page.js [hunk 17]
```diff
@@ -160,8 +160,11 @@ rxTimePicker.prototype.isOpen = function () {
...
 
 /**
+ * @deprecated
  * @description Whether the picker is closed
- * @returns {Boolean}
+ *
+ * **DEPRECATED** Check inverse of `isOpen()` instead
+ * @returns {Promise<Boolean>}
  */
 rxTimePicker.prototype.isClosed = function () {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxTimePicker.page.js]: ⬜ (Release Notes) DEPRECATED `rxTimePicker.isClosed()` Use `rxTimePicker.isOpen()` instead.

File: utils/rx-page-objects/src/rxToggleSwitch.page.js [hunk 18]
```diff
@@ -68,6 +68,8 @@ var rxToggleSwitch = {
      * @function
      * @instance
      * @description Whether or not the switch component is currently set to the "off" position.
+     *
+     * **DEPRECATED** Check inverse of `isEnabled()` instead.
      * @returns {Boolean}
      */
     isDisabled: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxToggleSwitch.page.js]: ⬜  (Release Notes) DEPRECATED `rxToggleSwitch.isDisabled()`. Use `rxToggleSwitch.isEnabled()` instead.

File: utils/rx-page-objects/src/rxToggleSwitch.page.js [hunk 19]
```diff
@@ -68,6 +68,8 @@ var rxToggleSwitch = {
      * @function
      * @instance
      * @description Whether or not the switch component is currently set to the "off" position.
+     *
+     * **DEPRECATED** Check inverse of `isEnabled()` instead.
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxToggleSwitch.page.js]: FYI: this function was already removed in 2.0.0-2

File: utils/rx-page-objects/src/rxForm.page.js [hunk 20]
```diff
@@ -25,11 +25,26 @@ var rxFieldName = {
...
+        }
+    },
+
+    /**
+     * @function
+     * @instance
+     * @description Whether or not a required field currently displays a red asterisk next to it.
+     * @returns {Boolean}
+     */
+    isSymbolDisplayed: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxForm.page.js]: more consistent with protractor/selenium naming

File: utils/rx-page-objects/src/rxTags.page.js [hunk 21]
```diff
@@ -32,15 +32,35 @@ var tag = function (tagElement) {
...
+         * @function
+         * @memberof rxTags.tag
+         * @description Whether or not the tag exists.
+         * @example
+         * it('should have a tag present after creating it', function () {
+         *     expect(encore.rxTags.initialize().addTag('Foo').isPresent()).to.eventually.be.true;
+         * });
+         * @returns {Promise<Boolean>}
+         */
+        isPresent: {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxTags.page.js]: consistent with protractor/selenium naming

File: utils/rx-page-objects/src/rxDatePicker.page.js [hunk 22]
```diff
@@ -120,17 +120,35 @@ var rxDatePicker = {
...
+
+    /**
+     * @function
+     * @instance
+     * @description Whether or not the entire calendar component is enabled.
+     * @return {Promise<Boolean>}
+     */
+    isEnabled: {
+        value: function () {
             return this.rootElement.getAttribute('disabled').then(function (disabled) {
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxDatePicker.page.js]: Can probably modify to `return this.rootElement.isEnabled()`

Comment (CITguy) [inline utils/rx-page-objects/src/rxDatePicker.page.js]: Protractor `isEnabled()` doesn't seem to work unless it's called on an HTML form control element. We'll leave this as is.

**Comment by 100stacks** [comment]
LGTM

**Comment by parlarjb** [comment]
LGTM

**Comment by CITguy** [comment]
@Droogans: any input?

File: utils/rx-page-objects/src/rxApp.page.js [hunk 23]
```diff
@@ -64,38 +64,41 @@ var rxApp = {
...
         value: function () {
-            return this.isCollapsed().then(function (result) {
-                return !result;
+            var page = this;
+            return this.isCollapsible().then(function (isCollapsible) {
+                if (!isCollapsible) {
+                    return page.title.then(function (siteTitle) {
+                        page.NotCollapsibleException.thro(siteTitle);
+                    });
+                }
```
Comment (Droogans) [inline utils/rx-page-objects/src/rxApp.page.js]: @CITguy do you want to see more or less of this kind of code?

Comment (CITguy) [inline utils/rx-page-objects/src/rxApp.page.js]: The question to answer is "Do we really want to get in the habit of adding exception logic to page objects?"

I'm inclined to do less of this.

Comment (Droogans) [inline utils/rx-page-objects/src/rxApp.page.js]: There is an argument to be had for "it's nicer to have a human-readable error message like `'You can't use this page object like this'` versus `'UnknownError: element not clickable at point (x, y)'`", but at this point I'm more than happy to update our page objects to no longer throw errors outside of the usual ones that you'd experience writing protractor code yourself. Less things to test, and only marginal benefits.

Comment (CITguy) [inline utils/rx-page-objects/src/rxApp.page.js]: We can discuss offline.

**Pull request merged**